recipes/boost/bld.bat

```diff
@@ -1,14 +1,18 @@
 :: Set the right msvc version according to Python version
-if "%PY_VER%"=="2.7" (
-    set MSVC_VER=9.0
-    set LIB_VER=90
-) else if "%PY_VER%"=="3.4" (
-    set MSVC_VER=10.0
-    set LIB_VER=100
-) else (
-    set MSVC_VER=14.0
-    set LIB_VER=140
-)
+REM write a temporary batch file to map cl.exe version to visual studio version
+echo @echo 15=9 > msvc_versions.bat
+echo @echo 16=10 >> msvc_versions.bat
+echo @echo 17=11 >> msvc_versions.bat
+echo @echo 18=12 >> msvc_versions.bat
+echo @echo 19=14 >> msvc_versions.bat
+
+REM Run cl.exe to find which version our compiler is
+for /f "delims=" %%A in ('cl /? 2^>^&1 ^| findstr /C:"Version"') do set "CL_TEXT=%%A"
+FOR /F "tokens=1,2 delims==" %%i IN ('msvc_versions.bat') DO echo %CL_TEXT% | findstr /C:"Version %%i" > nul && set VSTRING=%%j && goto FOUND
+EXIT 1
+:FOUND
+
+call :TRIM VSTRING %VSTRING%
```

Comment:
It is set with that for loop.  This is very strange syntax, but that is how Bat files get output from programs.